### PR TITLE
Sy 1078 docssite meta data documentation

### PR DESCRIPTION
**Pull request by emilbon99**
# Feature Pull Request Template

## Key Information

- **Linear Issue**: [SY-1078](https://linear.app/synnax/issue/SY-1078/[docssite]-meta-data-documentation)

## Description

Maximum 2-3 sentence description describing the changes.

## Basic Readiness Checklist

- [x] I have performed a self-review of my code.
- [x] I have added relevant tests to cover the changes to CI.
- [x] I have updated user facing documentation accordingly.
- [x] I have verified code coverage targets are met.

## Migrations

- [ ] Console - I have ensured that previous versions of stored data structures are 
properly migrated to new formats.
- [ ] Server - I have ensured that previous versions of stored data structures are 
properly migrated to new formats.

##  Additional Notes
- [ ] These changes deal with concurrency.
- [ ] These changes affect UI.

## Manual QA Additions

- [ ] I have updated the [Release Candidate](/.github/PULL_REQUEST_TEMPLATE/rc.md) template
with necessary manual QA steps to test my change.

## Breaking Changes

Please list any breaking changes to public or internal packages.

## Reviwer Checklist
- [ ] Sufficient test coverage of new additions.
- [ ] Verified all steps in the Readiness checklists.
- [ ] UI changes have been tested.
- [ ] Style and formatting is consistent.
- [ ] Reviewed any relevant changes to concurrent code for safety. 
- [ ] Sufficient comments and clarity of code.
- [ ] Any additional documentation necessary for feature is provided and has been reviewed for clarity.

### CODE REVIEW

File: client/py/examples/dev/bluefors.py
Comment (Lham42) [inline client/py/examples/dev/bluefors.py:1]: remove

File: client/py/examples/dev/pdu.py
Comment (Lham42) [inline client/py/examples/dev/pdu.py:1]: remove

File: console/src/main.tsx [hunk 1]
```diff
@@ -9,8 +9,8 @@
 
 import "@/index.css";
 import "@synnaxlabs/media/dist/style.css";
-import "@synnaxlabs/pluto/dist/style.css";
 
+// import "@synnaxlabs/pluto/dist/style.css";
```
Comment (Lham42) [inline console/src/main.tsx:13]: can this be deleted?

File: docs/site/.astro/types.d.ts
Comment (Lham42) [inline docs/site/.astro/types.d.ts:1]: is this file just one line now? can we remove it

File: docs/site/src/components/Rule.astro [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
...8.5%22 cy=%221390.5%22 r=%224%22 fill=%22%23434343a9%22%2F%3E%3Ccircle cx=%221255.5%22 cy=%221390.5%22 r=%224%22 fill=%22%23434343b6%22%2F%3E%3Ccircle cx=%221309.5%22 cy=%221390.5%22 r=%224%22 fill=%22%2343434371%22%2F%3E%3Ccircle cx=%221363.5%22 cy=%221390.5%22 r=%224%22 fill=%22%23434343f0%22%2F%3E%3Ccircle cx=%221444.5%22 cy=%221390.5%22 r=%224%22 fill=%22%23434343e5%22%2F%3E%3Ccircle cx=%221525.5%22 cy=%221390.5%22 r=%224%22 fill=%22%2343434362%22%2F%3E%3Ccircle cx=%221579.5%22 cy=%221390.5%22 r=%224%22 fill=%22%23434343f8%22%2F%3E%3Ccircle cx=%221660.5%22 cy=%221390.5%22 r=%224%22 fill=%22%2343434375%22%2F%3E%3Ccircle cx=%221822.5%22 cy=%221390.5%22 r=%224%22 fill=%22%234343432e%22%2F%3E%3Ccircle cx=%221849.5%22 cy=%221390.5%22 r=%224%22 fill=%22%2343434367%22%2F%3E%3Ccircle cx=%221957.5%22 cy=%221390.5%22 r=%224%22 fill=%22%23434343cf%22%2F%3E%3Ccircle cx=%222011.5%22 cy=%221390.5%22 r=%224%22 fill=%22%23434343e0%22%2F%3E%3C%2Fg%3E%3C%2Fg%3E%3Cdefs%3E%3CradialGradient id=%22a%22%3E%3Cstop offset=%220%22 stop-color=%22%23fff%22 stop-opacity=%220%22%2F%3E%3Cstop offset=%221%22 stop-color=%22%23fff%22 stop-opacity=%22.6%22%2F%3E%3C%2FradialGradient%3E%3C%2Fdefs%3E%3C%2Fsvg%3E");
```
Comment (Lham42) [inline docs/site/src/components/Rule.astro:23]: this url is meant to be like this?

File: docs/site/src/components/search/Search.tsx [hunk 3]
```diff
@@ -55,6 +152,7 @@ export const Search = (): ReactElement => {
       }),
     });
     const json = await res.json();
+    console.log(json);
```
Comment (Lham42) [inline docs/site/src/components/search/Search.tsx:155]: can we remove

File: docs/site/src/components/search/Search.tsx [hunk 4]
```diff
@@ -41,10 +43,105 @@ const ALGOLIA_HEADERS = {
...
+      if (part.length === 0 || index === parts.length - 1) return null;
+      const split = part
+        .split("-")
+        .filter((p) => p.length > 0)
+        .map(caseconv.capitalize);
+      return split.join(" ");
+    },
+    "/",
+  );
+  console.log(path);
```
Comment (Lham42) [inline docs/site/src/components/search/Search.tsx:112]: can we remove

File: docs/site/src/pages/reference/concepts/channels.mdx [hunk 5]
```diff
@@ -14,86 +18,102 @@ visit the respective pages for the [Python](../python-client/channels) and
 
 <Divider.Divider direction="x" />
 
-## What is a Channel?
+## Introduction
 
 A channel is a logical collection of samples emitted by or representing the
```
Comment (Lham42) [inline docs/site/src/pages/reference/concepts/channels.mdx:23]: can we just say "A channel is a logical grouping of samples that represent the values from a single source". not sure the emitted by or representing part really adds more clarity

File: docs/site/src/pages/reference/typescript-client/write-data.mdx [hunk 6]
```diff
@@ -13,6 +13,12 @@ The Synnax TypeScript client supports multiple methods for writing data to a
 cluster. We can write directly to a channel, or we can write to multiple
 channels in a streaming fashion using a writer.
 
+Write in Synnax are more complicated than reads, and, as such, we
```
Comment (Lham42) [inline docs/site/src/pages/reference/typescript-client/write-data.mdx:16]: "write in synnax" typo?

File: docs/site/src/pages/reference/typescript-client/write-data.mdx [hunk 7]
```diff
@@ -13,6 +13,12 @@ The Synnax TypeScript client supports multiple methods for writing data to a
 cluster. We can write directly to a channel, or we can write to multiple
 channels in a streaming fashion using a writer.
 
+Write in Synnax are more complicated than reads, and, as such, we 
+recommend checking out our [concepts](/reference/concepts/writes) apge
```
Comment (Lham42) [inline docs/site/src/pages/reference/typescript-client/write-data.mdx:17]: typo on apge

File: pluto/examples/package.json [hunk 8]
```diff
@@ -10,7 +10,7 @@
     "preview": "vite preview"
   },
   "dependencies": {
-    "@synnaxlabs/pluto": "^0.21.0",
+    "@synnaxlabs/pluto": "workspace:*",
```
Comment (Lham42) [inline pluto/examples/package.json:13]: this is intentional?

File: pluto/examples/vite.config.ts [hunk 9]
```diff
@@ -7,10 +7,18 @@
...
 // https://vitejs.dev/config/
 export default defineConfig({
   plugins: [react()],
-})
+  resolve: {
+    // alias: {
+    //   "@synnaxlabs/pluto/dist": path.resolve(__dirname, "../dist"),
+    //   "@synnaxlabs/pluto": path.resolve(__dirname, "../src"),
+    //   "@synnaxlabs/drift/dist": path.resolve(__dirname, "../dist"),
+    // },
```
Comment (Lham42) [inline pluto/examples/vite.config.ts:22]: do we still need this commented out code

**Review by Lham42 — CHANGES_REQUESTED** [review]

File: console/src/main.tsx (hunk 1 above)
Comment (emilbon99) [reply console/src/main.tsx:13]: No. In fact its critical that this gets commented back in.

File: docs/site/.astro/types.d.ts
Comment (emilbon99) [reply docs/site/.astro/types.d.ts:1]: It is just one line. No, we can't remove it. It's auto-generated and astro needs it to function.

File: docs/site/src/components/Rule.astro (hunk 2 above)
Comment (emilbon99) [reply docs/site/src/components/Rule.astro:23]: Yes, it gives the background a cool pattern

File: docs/site/src/pages/reference/concepts/channels.mdx (hunk 5 above)
Comment (emilbon99) [reply docs/site/src/pages/reference/concepts/channels.mdx:23]: Yes. Corrected

File: docs/site/src/pages/reference/typescript-client/write-data.mdx (hunk 6 above)
Comment (emilbon99) [reply docs/site/src/pages/reference/typescript-client/write-data.mdx:16]: Fixed

File: pluto/examples/package.json (hunk 8 above)
Comment (emilbon99) [reply pluto/examples/package.json:13]: Yes

File: pluto/examples/vite.config.ts (hunk 9 above)
Comment (emilbon99) [reply pluto/examples/vite.config.ts:22]: It should be commented back in

File: docs/site/src/components/search/Search.tsx (hunk 3 above)
Comment (emilbon99) [reply docs/site/src/components/search/Search.tsx:155]: Yes

File: docs/site/src/components/search/Search.tsx (hunk 4 above)
Comment (emilbon99) [reply docs/site/src/components/search/Search.tsx:112]: And yes

File: docs/site/.astro/types.d.ts
Comment (Lham42) [reply docs/site/.astro/types.d.ts:1]: ok

File: console/src/main.tsx (hunk 1 above)
Comment (Lham42) [reply console/src/main.tsx]: so why is it not commented in right now

**Review by Lham42 — APPROVED** [review]

**Review by Lham42 — CHANGES_REQUESTED** [review]
@emilbon99 console build failing

**Comment by emilbon99** [comment]
@Lham42 Fixed

File: docs/site/src/pages/reference/typescript-client/write-data.mdx [hunk 10]
```diff
@@ -13,11 +13,17 @@ The Synnax TypeScript client supports multiple methods for writing data to a
 cluster. We can write directly to a channel, or we can write to multiple
 channels in a streaming fashion using a writer.
 
+Writes in Synnax are more complicated than reads, and, as such, we 
+recommend checking out our [concepts](/reference/concepts/writes) page
+to learn more about writing data to Synnax.
+
+At the very least, you should be familiar with the [rules of writes](/reference/concepts/writes#rules-of-writes).
```
Comment (pjdotson) [inline docs/site/src/pages/reference/typescript-client/write-data.mdx:20]: Same comment as Python client.

File: docs/site/src/pages/reference/python-client/write-data.mdx [hunk 11]
```diff
@@ -12,14 +12,16 @@ directly to a channel, fetch a range and set its data, or leverage writers for
 writing large volumes of data.
 
 Writes in Synnax are more complicated than reads, and, as such, we recommend
-checking out our [concepts](../concepts/write) page to learn more about
-the best practices for writing data to Synnax.
+checking out our [concepts](/reference/concepts/write) page to learn more about
+the best practices for writing data to Synnax. 
+
+At the very least, you should be familiar with the [rules of writes](/reference/concepts/writes#rules-of-writes).
```
Comment (pjdotson) [inline docs/site/src/pages/reference/python-client/write-data.mdx:18]: The tone on this statement is off-putting and somewhat condescending. Why don't you just add the sentence "The rules of writes are especially important." to the end of the previous paragraph?

File: docs/site/src/pages/reference/console/get-started.mdx [hunk 12]
```diff
@@ -7,6 +7,7 @@ description: "Install and use the Synnax Console."
 import { Align, Divider } from "@synnaxlabs/pluto";
 import { SynnaxConsoleDownloadButton } from "@/components/console/OSDownloadButton";
 import { Image } from "@/components/Media";
+import Diagram from "@/components/Diagram.astro";
```
Comment (pjdotson) [inline docs/site/src/pages/reference/console/get-started.mdx:10]: This isn't used

File: docs/site/src/pages/reference/concepts/streams.mdx [hunk 13]
```diff
@@ -6,6 +6,9 @@ description: "Learn the process for streaming data from a Synnax cluster."
 ---
 
 import { Divider } from "@synnaxlabs/pluto";
+import { Image } from "@/components/Media";
```
Comment (pjdotson) [inline docs/site/src/pages/reference/concepts/streams.mdx:9]: This import is not used

File: docs/site/src/pages/reference/concepts/_nav.tsx [hunk 14]
```diff
@@ -43,5 +43,10 @@ export const conceptsNav: PageNavNode = {
       href: "/reference/concepts/streams",
       name: "Streams",
     },
+    {
```
Comment (pjdotson) [inline docs/site/src/pages/reference/concepts/_nav.tsx:46]: I think ranges should come between channels and reads instead of at the end.

**Review by pjdotson — CHANGES_REQUESTED** [review]
Left a lot of comments here and there are some failing CI checks that need to be addressed. Another thing that we should do is make a better way for moving to subsequent pages. See how on the bottom of the react page we have the previous / next navigator? If you don't want to do this in this PR then an issue should be made for it.
image

**Comment by emilbon99** [comment]
> Left a lot of comments here and there are some failing CI checks that need to be addressed. Another thing that we should do is make a better way for moving to subsequent pages. See how on the bottom of the react page we have the previous / next navigator? If you don't want to do this in this PR then an issue should be made for it. image

Yeah let's turn this into a separate issue

**Pull request merged**